app/gui2/src/stores/graph/index.ts

```diff
@@ -418,7 +418,7 @@ export const useGraphStore = defineStore('graph', () => {
   }
 
   function getPortNodeId(id: PortId): NodeId | undefined {
-    return getPortPrimaryInstance(id)?.nodeId ?? db.getExpressionNodeId(id as string as Ast.AstId)
+    return db.getExpressionNodeId(id as string as Ast.AstId) ?? getPortPrimaryInstance(id)?.nodeId
```

Comment:
I'm not sure about this change. If we have port instances, and they define their nodeId, they could have a reason to change it from AST?

Reply:
They don’t really “define” their nodeid, they use the node id derived from AST. In which case will they change it?

Reply:
so, the node id is derived from AST at some point, but not necessarily from _Port_ Id. PortId is a broader than just AstId. Technically (even if it's an awful idea), some widget could create portIds which also resembles AstId, but are not bound to them. In that case, I would say the port information from PortId is more accurate than what we deduce from AstId. `getExpressionNodeId` feels more like a fallback than the main route.